File: sdk/python/src/claude_code_rs/claude_code.py
"""Top-level client — analogous to ``Codex`` in the Codex SDK.

Mirrors TypeScript SDK's claudeCode.ts.

Usage::

    from claude_code_rs import ClaudeCode, SessionOptions

    client = ClaudeCode()
    session = client.start_session(SessionOptions(model="claude-sonnet-4-20250514"))
    turn = session.run("What files are in this directory?")
    print(turn.final_response)
"""

from __future__ import annotations

from dataclasses import replace

from .config import ClientOptions, SessionOptions
from .exec import ClaudeCodeExec
from .session import Session


class ClaudeCode:
    """Entry point for the claude-code-rs Python SDK."""

    def __init__(self, options: ClientOptions | None = None) -> None:
        self._options = options or ClientOptions()
        self._exec = ClaudeCodeExec(
            executable_path=self._options.executable_path,
            env=self._options.env,
        )

    def start_session(
        self, options: SessionOptions | None = None
    ) -> Session:
        """Create a new session."""
        return Session(
            self._exec,
            self._options,
            options or SessionOptions(),
        )

    def resume_session(
        self,
        session_id: str,
        options: SessionOptions | None = None,
    ) -> Session:
        """Resume a previously persisted session by ID."""
        opts = options or SessionOptions()
        merged = replace(opts, continue_session=session_id)
        return Session(self._exec, self._options, merged)

File: sdk/typescript/src/claudeCode.ts
/**
 * Top-level client — analogous to `Codex` in the Codex SDK.
 *
 * Usage:
 * ```ts
 * const client = new ClaudeCode();
 * const session = client.startSession({ model: "claude-sonnet-4-20250514" });
 * const turn = await session.run("What files are in this directory?");
 * console.log(turn.finalResponse);
 * ```
 */

import type { ClaudeCodeOptions } from "./claudeCodeOptions.js";
import { ClaudeCodeExec } from "./exec.js";
import { Session } from "./session.js";
import type { SessionOptions } from "./sessionOptions.js";

export class ClaudeCode {
  private exec: ClaudeCodeExec;
  private options: ClaudeCodeOptions;

  constructor(options: ClaudeCodeOptions = {}) {
    this.exec = new ClaudeCodeExec(
      options.executablePath ?? null,
      options.env,
    );
    this.options = options;
  }

  /** Create a new session. */
  startSession(options: SessionOptions = {}): Session {
    return new Session(this.exec, this.options, options);
  }

  /** Resume a previously persisted session by ID. */
  resumeSession(sessionId: string, options: SessionOptions = {}): Session {
    return new Session(this.exec, this.options, {
      ...options,
      continueSession: sessionId,
    });
  }
}
